src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_CAPACITY = 10;
+    private T[] array;
+    private int sizeOfFilledArray;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_ARRAY_CAPACITY];
+        this.sizeOfFilledArray = 0;
+    }
+
+    public void growCapacity() {
```

Comment:
let’s move private methods to the bottom of the file, or at least after the place where we call them